src/main/java/com/cleanroommc/groovyscript/event/EventHandler.java

```diff
@@ -4,20 +4,51 @@
 import com.cleanroommc.groovyscript.api.GroovyLog;
 import com.cleanroommc.groovyscript.compat.vanilla.CraftingInfo;
 import com.cleanroommc.groovyscript.compat.vanilla.ICraftingRecipe;
+import com.cleanroommc.groovyscript.compat.vanilla.Player;
 import com.cleanroommc.groovyscript.core.mixin.InventoryCraftingAccess;
 import com.cleanroommc.groovyscript.core.mixin.SlotCraftingAccess;
 import com.cleanroommc.groovyscript.sandbox.ClosureHelper;
 import groovy.lang.Closure;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.inventory.*;
+import net.minecraft.item.ItemStack;
 import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.nbt.NBTTagCompound;
 import net.minecraftforge.fml.common.Mod;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.minecraftforge.fml.common.gameevent.PlayerEvent;
+import net.minecraftforge.items.ItemHandlerHelper;
 
 @Mod.EventBusSubscriber(modid = GroovyScript.ID)
 public class EventHandler {
 
+    @SubscribeEvent
+    public static void playerLogin(PlayerEvent.PlayerLoggedInEvent event) {
+        NBTTagCompound tag = event.player.getEntityData();
+        NBTTagCompound data;
+        if (!tag.hasKey(EntityPlayer.PERSISTED_NBT_TAG)) {
+            data = new NBTTagCompound();
+        } else {
+            data = tag.getCompoundTag(EntityPlayer.PERSISTED_NBT_TAG);
+        }
+        if (!data.getBoolean(Player.givenItems)) {
+            Player.itemMap.forEach((stack, slot) -> {
+                if (slot <= -1) {
+                    ItemHandlerHelper.giveItemToPlayer(event.player, stack.copy());
+                } else {
+                    if (event.player.inventory.getStackInSlot(slot) != ItemStack.EMPTY) {
+                        ItemHandlerHelper.giveItemToPlayer(event.player, stack.copy());
+                    } else {
+                        event.player.inventory.setInventorySlotContents(slot, stack.copy());
```

Comment:
Better would be to use vanilla's `InventoryPlayer#add/addItemStackToInventory` as it handles the `-1` slot already.

```java
Player.itemMap.forEach((stack, slot) -> player.inventory.add(slot, stack.copy())
```